### DISCUSSION TITLE
Why is SPI mode not supported for Raspberry Pi Pico?

### DISCUSSION THREAD

**Discussion by DoomHammer** [board707/DMD_STM32#117] [Q&A]
The title says it all. Is it impossible to implement or just hasn't been implemeted yet?

**Comment by board707** [comment]
Hi
To be honest, I didn't planned SPI mode for Rasberry Pico, because Parallel mode is fast and does not load the main controller when using PIO machines and DMA.  As about memory overhead when using parallel mode - it is not critical for RP 2040 mcu.

If you have any arguments why it is necessary, I will be happy to read them.

**Reply by DoomHammer** [reply]
I'm not sure if I fully understand what parallel means in this context. I want to connect 8 panels. Does it mean I have to use a single pin each for every module? Or do I connect Pico to the first module and then IDC to connect the subsequent ones?

**Reply by board707** [reply]
It depends on configuration of your screen. The general rule is that you need to connect every horizontal row of panels to separate pin.
So if your screen is a single 8 panel row - connect all panels to first pin the same way as with SPI mode.
If your screen configuration is 4x2 - connect upper row of 4 panels to first pin and lower row - to second one.

**Reply by DoomHammer** [reply]
OK. Now I get it. Worked like a charm :)

Thanks!
